### ISSUE TITLE
"Source URL must not be empty" for new project after recent Ubuntu on wsl update

### ISSUE THREAD

**Issue by shylane** [pdm-project/pdm#2072]
- [x] I have searched the issue tracker and believe that this is not a duplicate.

## Steps to reproduce

I just now updated my Ubuntu on WSL. Then created a new project. And then found the following error.
I had created projects earlier as well without any such issue. Can anyone please help?

## Actual behavior

```console
xyz@abcd:~/PDM/google_cloud_1$ pdm info -v
Traceback (most recent call last):
  File "/home/xyz/.local/bin/pdm", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/xyz/.local/share/pdm/venv/lib/python3.11/site-packages/pdm/core.py", line 289, in main
    return Core().main(args)
           ^^^^^^^^^^^^^^^^^
  File "/home/xyz/.local/share/pdm/venv/lib/python3.11/site-packages/pdm/core.py", line 207, in main
    raise cast(Exception, err).with_traceback(traceback) from None
  File "/home/xyz/.local/share/pdm/venv/lib/python3.11/site-packages/pdm/core.py", line 202, in main
    self.handle(project, options)
  File "/home/xyz/.local/share/pdm/venv/lib/python3.11/site-packages/pdm/core.py", line 157, in handle
    command.handle(project, options)
  File "/home/xyz/.local/share/pdm/venv/lib/python3.11/site-packages/pdm/cli/commands/info.py", line 32, in handle
    interpreter = project.environment.interpreter
                  ^^^^^^^^^^^^^^^^^^^
  File "/home/xyz/.local/share/pdm/venv/lib/python3.11/site-packages/pdm/project/core.py", line 280, in environment
    self._environment = self.get_environment()
                        ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/xyz/.local/share/pdm/venv/lib/python3.11/site-packages/pdm/project/core.py", line 258, in get_environment
    PythonEnvironment(self)
  File "/home/xyz/.local/share/pdm/venv/lib/python3.11/site-packages/pdm/environments/python.py", line 17, in __init__
    super().__init__(project, python=python)
  File "/home/xyz/.local/share/pdm/venv/lib/python3.11/site-packages/pdm/environments/base.py", line 42, in __init__
    self.auth = PdmBasicAuth(project.core.ui, self.project.sources)
                                              ^^^^^^^^^^^^^^^^^^^^
  File "/home/xyz/.local/share/pdm/venv/lib/python3.11/site-packages/pdm/project/core.py", line 383, in sources
    assert source.url, "Source URL must not be empty"
AssertionError: Source URL must not be empty
```

## Environment Information

```console
xyz@abcd:~/PDM/google_cloud_1$ pdm info
PDM version:
  2.7.4
Python Interpreter:
  /home/xyz/PDM/google_cloud_1/.venv/bin/python (3.11)
Project Root:
  /home/xyz/PDM/google_cloud_1
Local Packages:
```

Release: Ubuntu 23.04
Kernel: Linux 5.15.90.1-microsoft-standard-WSL2

**Comment by pawamoy** [comment]
Can you check your pyproject.toml and PDM configuration file (`~/.config/pdm/config.toml` on Linux) to assert that you don't have any empty source in them?

**Comment by shylane** [comment]
There is no source field in either files.
But my previous project was working fine the same way without such a field in any of those files.

**Comment by frostming** [comment]
> There is no source field in either files.

please check if there are any empty `[pypi.<name>]` tables there. Also this should be fixed on the 2.8.0 prerelease by #2034

**Comment by shylane** [comment]
Yeah, indeed there was an empty [pypi.extra].

Must be because a library I tried installing and then removed eventually.

Clearing that entry, fixed it. Thank you.
